src/main/java/org/opentripplanner/model/impl/OtpTransitServiceBuilder.java

```diff
@@ -111,37 +105,53 @@ public OtpTransitServiceBuilder add(OtpTransitService other) {
         return pathways;
     }
 
-    public List<Route> getRoutes() {
-        return routes;
+    public EntityMap<FeedScopedId, Route> getRoutes() {
+        return routesById;
     }
 
     public List<ShapePoint> getShapePoints() {
         return shapePoints;
     }
 
-    public List<Stop> getStops() {
-        return stops;
+    public EntityMap<FeedScopedId, Stop> getStops() {
```

Comment:
It feels wrong to me to have getters on a Builder class. I realize this has been adapted from a pre-existing class that had getters. But I see several of these getters being used to fetch a collection, on which the caller then calls add(). It would make more sense to replace these getters with methods like addStops() or addRoutes().

However, I see that in some places like GtfsContextBuilder.build() we are actually fetching collections from inside this Builder and modifying them. I see that this is just a way of progressively refactoring existing code, but it seems like it could be further refactored to have more conventional behavior.

Reply:
I just tried to be consistent, but I see your point. I think is is vice to leave it for now and return to it after we have applied the NeTEx import - which (I am afraid) takes this problem to a whole new level (-; 

For the record, there are several solutions:
 - making immutable access til list and add explicit add methods
 - making a custom "List" list like the EntityMap that have methods for adding and immutable iteration.
 - Also, maybe the "builder/service" names is wrong - and we should spend some time modeling this properly.

Reply:
Consistency is of course an important goal as well. But in this case it seems like getters on a builder are exceptions - necessary deviations from an ideal model. So maybe the getter should have mostly additive / constructive functions, and only a few exceptional getters where the current design happens to require it. 

I agree with your three points here. Both of the first two could make the code easier to reason about, and it's interesting to consider that Builder is simply not the right name/concept here. Actually we've got more of a mutable and immutable version of the same object. When I look at it that way, the boilerplate repetition seems kind of excessive. The existence of mutable builders and immutable output classes from builders seems better justified to me, in that the Builder also provides a convenient fluent API for object preparation.

Let's rethink only after we've got the existing Netex handlers merged in.